pkg/config/cni.go

```diff
@@ -50,28 +65,60 @@ func MigrateCNIConfig(ctx context.Context, fullState *cluster.FullState, dataDir
 				"iface":   networkConfig.Options[canalFlannelInterface],
 			},
 		}
-		helmChartConfig, err := toHelmChartConfig("rke2-"+canalCNI, canalCfg)
+		helmChartConfig, err = toHelmChartConfig("rke2-"+networkConfig.Plugin, canalCfg)
```

Comment:
Does the `Plugin` struct implement `Stringer`? A cursory search on my part didn't turn up a hit. Even if it does, seems like unreliable magic.

Reply:
Hmm, its not a struct. Sourcegraph is punching me in the throat here.

Reply:
It appears to be a string.
https://github.com/rancher/rke/blob/22b82828ffa0ddfe4a45198da553fe7e6071d01f/types/rke_types.go#L383

Reply:
no this is the networkConfig of rke1 type, it should be here https://github.com/rancher/rke/blob/master/types/rke_types.go#L383